tests/plugins/test_ddp_plugin_with_comm_hook.py

```diff
@@ -108,3 +110,33 @@ def test_ddp_spawn_fp16_compress_comm_hook(tmpdir):
     )
     trainer.fit(model)
     assert trainer.state.finished, f"Training failed with {trainer.state}"
+
+
+@RunIf(skip_windows=True, min_torch="1.9.0", min_gpus=2, special=True)
+def test_ddp_post_local_sgd_comm_hook(tmpdir):
+    """Test for DDP post-localSGD hook."""
+    model = BoringModel()
+
+    training_type_plugin = DDPPlugin(
+        ddp_comm_state=post_localSGD.PostLocalSGDState(
+            process_group=None,
+            subgroup=None,
+            start_localSGD_iter=8,
+        ),
+        ddp_comm_hook=post_localSGD.post_localSGD_hook,
+        model_averaging_period=4,
+        sync_batchnorm=True,
```

Comment:
is this strictly necessary? if so, will a user know?

Reply:
I don't know actually. I actually just followed the other examples above. Pls let me know if you have a better suggestion. Thx!

Reply:
hmm ok, I see. 
then let's keep it. I guess we want to test that they work well together. on the other hand, if there were issues, it would probably be on the pytorch side. 
so either way is fine I think.